src/generators/generator.py

```diff
@@ -544,14 +556,29 @@ def _add_node_to_parent(self, parent_namespace, node):
             ast.ParameterDeclaration: self.context.add_var,
             ast.Lambda: self.context.add_lambda,
         }
-        if parent_namespace == ast.GLOBAL_NAMESPACE:
+        lang_specific_pairs = self.bt_factory.update_add_node_to_parent()
```

Comment:
I think we should follow a similar approach as above to eliminate the following, dirty if condition.